reactive/spark.py

```diff
@@ -53,38 +54,97 @@ def install_spark(hadoop):
         set_state('spark.installed')
 
 
-@when('spark.installed', 'hadoop.ready')
+@when('hadoop.ready', 'spark.installed')
+@when_not('livy.installed')
+def install_livy(hadoop):
+    dist = DistConfig(data=layer.options('livy'))
+    livy = Livy(dist)
+    if livy.verify_resources():
+        hookenv.status_set('maintenance', 'Installing Livy REST server')
+        dist.add_users()
```

Comment:
I prefer adding uses/groups/directories to be handled inside the Livy/Spark class. The reason is that the Livy/Spark objects consume the DistConfig and know if they need users/dirs/groups